rfcs/ocamlib.md

```diff
@@ -473,8 +612,67 @@ code base.
 
 Starting from Dune 4, Dune will no longer generate `META` files.
 
+## Examples
+
+### Compiling and using libraries locally in a build
+
+The following shows how to compile and layout two libraries 
+to use locally in a build. We have:
+
+* Library `a` made of source `a.ml` depending on the `ocaml.str` 
+  library.
+* Library `b` made of source `b.ml` depending on the library `a`.
+* An executable `exec.ml` which depends on `b`
+
+Here are our sources:
+
+```
+.
+├── a.ml
+├── b.ml
+└── exec.ml
+```
+We create directories for the libraries. The `libs` directory 
+will be added to the `OCAMLPATH`.
+
+```
+mkdir -p libs libs/a libs/b
+```
+
+We compile and layout library `a`:
+
+```
+ocamlopt -require ocaml.str -c -o libs/a/a.cmx a.ml
+ocamlopt -require ocaml.str -a -o libs/a/lib.cmxa libs/a/a.cmx
+```
+
+We compile and layout library `b`. We extend the OCAMLPATH on the 
+command line via the `-L` option.
+
+```
+ocamlopt -L libs -require a -c -o libs/b/b.cmx b.ml
+ocamlopt -L libs -require a -a -o libs/b/lib.cmxa libs/b/b.cmx
+```
+
+We compile our executable:
+```
+ocamlopt -L libs -require b exec.ml
+```
+
 ## Unresolved issues 
 
+### OCAMLPATH initial value
+
+At the moment the proposal has no default value for the `OCAMLPATH`.
+This means that without specifying any `OCAMLPATH` only the stdlib is
+available for compiling and by definition any `-require LIB` will
+fail for any library name `LIB`.
+
+It is expected that system itegrators will setup an appropriate
+`OCAMLPATH` when needed (e.g. on `eval $(opam env)`) and according to
+the install structure they choose. Whether a more complex mecanism is
+needed will be discussed in due time with system integrators.
```

Comment:
Here's a solution that could be used.

1. Have an optional configure option that gives a default value to `OCAMLPATH`. This value is only used if the environment variable `OCAMLPATH` is *undefined* (vs undefined or empty). This configure value is not guessed
and left empty if unspecified.
2. Add `ocamlc -ocamlpath` which outputs, using the platform convention, the current effective value of the `OCAMLPATH` which is either the value of the `OCAMLPATH` variable if defined or the the configure value if undefined.

This means that people who want to extend OCAMLPATH and keep the configure default can do it with:
```
export OCAMLPATH=mydir:$(ocamlc -ocamlpath)
```

This also means that the system is easy to understand: 

1. If the `OCAMLPATH` variable is undefined the OCAMLPATH is the value output by `ocamlc -ocamlpath`
2. If the `OCAMLPATH` variable is defined the OCAMLPATH is exactly the value of the `OCAMLPATH` variable (and incidentally also the value of `ocamlc -ocamlpath`)

Note that this is only expected to be used by system integrators. Things like `opam` will likely prefer to setup an `OCAMLPATH` on `opam env` (if only because it simplifies relocation).

@bobot thoughts ?

Reply:
In a private discussion @bobot mentioned that some tools have the convention that a final empty directory appends the default path. This allows to extend the default path by writing: 
```
OCAMLPATH=mydir:$OCAMLPATH
```
(e.g. `TEXINPUTS` works what way). 

Personally I find that to be a bit too subtle and obscure this means that: 
```
OCAMLPATH=dir
OCAMLPATH=dir:
```
have vastly non-obvious different semantics. I prefer the current text which says that any empty path is in `OCAMLPATH` is discarded. Besides with the solution outlined in the previous comment a reading of `OCAMLPATH`, if defined, always tells you exactly the whole picture which I find to be a good property. I amended the text in 4dd989d5434af174397 with that solution. We'll still need to ask system packagers if that works for them at some point.